cmd/ocsp-updater/main.go

```diff
@@ -212,49 +414,64 @@ func main() {
 		blog.SetAuditLogger(auditlogger)
 
 		go cmd.DebugServer(c.OCSPUpdater.DebugAddr)
+		go cmd.ProfileCmd("OCSP-Updater", stats)
 
 		// Configure DB
 		dbMap, err := sa.NewDbMap(c.OCSPUpdater.DBConnect)
 		cmd.FailOnError(err, "Could not connect to database")
 
-		cac, closeChan := setupClients(c, stats)
+		cac, pubc, closeChan := setupClients(c, stats)
+
+		updater, err := newUpdater(
+			auditlogger,
+			stats,
+			clock.Default(),
+			dbMap,
+			cac,
+			pubc,
+			// Necessary evil for now
+			len(c.Publisher.CT.Logs),
+			c.OCSPUpdater,
+		)
+		cmd.FailOnError(err, "Failed to create updater")
 
+		auditlogger.Info(app.VersionString())
+
+		// Handle AMQP disconnections gracefully (and block other operations while
+		// disconnected...)
 		go func() {
-			// Abort if we disconnect from AMQP
 			for {
-				for err := range closeChan {
-					auditlogger.Warning(fmt.Sprintf(" [!] AMQP Channel closed, aborting early: [%s]", err))
-					panic(err)
-				}
+				err := <-closeChan
+				auditlogger.Warning(fmt.Sprintf(" [!] AMQP Channel closed, will reconnect in 5 seconds: [%s]", err))
+				time.Sleep(5 * time.Second)
+				updater.rpcMu.Lock()
```

Comment:
@rolandshoemaker, last we talked you mentioned that for programs that are both RPC clients and servers, the reconnect is already handled in `amqp-rpc.go`, and this program is unusual because it is only a client, not a server.

However, by my reading of the code in `amqp-rpc.go`, it appears that there is reconnect logic for RPC servers, but there is none for RPC clients, even if they coexist with an RPC server. That means this reconnect logic should really go in the client code in `amqp-rpc.go`, right? If so, please open a separate PR to do just that.